src/Exporter.php

```diff
@@ -447,10 +439,20 @@ public function processProductData($productsData)
 
         $this->trackSkippedProducts(array_diff($itemIds, $validItemIds));
 
-        foreach ($validItemIds as $itemId) {
-            $product = $this->createProductItem($productsData[$itemId]);
+        $productsCount = $this->getProductsStreamer->getProductsCount($stream);
+
+        if ($productsCount <= 0) {
+            return $this;
+        }
+
+        $trackSkippedProductsIds = [];
+
+        for ($i = 0; $i < $productsCount; $i++) {
+            $product = $this->createProductItemFromStream($stream, $i);
```

Comment:
$i is used to identify the position of the product in the stream, for future parsing.